drivers/usb-common.c

```diff
@@ -410,3 +410,80 @@ void warn_if_bad_usb_port_filename(const char *fn) {
 		__func__, fn);
 	return;
 }
+
+/* API neutral, handles retries */
+static int nut_usb_get_string_descriptor(
+	usb_dev_handle *udev,
+	int StringIdx,
+	int langid,
+	char *buf,
+	size_t buflen)
+{
+	int ret;
```

Comment:
it seems we need to init value to pass CI
`int ret = -1;`

Reply:
Thanks! I am not able to see the details of the CI without granting a third-party company write access to my repositories! No way! It is ridiculous because all details of the CI are public. I also wonder why this CI of doom runs 4 hours when it already finds a blocker early on.

As you can see from the code, there is no possibility of "ret" being used uninitialized, so there is absolutely no need to initialize it explicitly. But I am not going to fix the CI, so I am applying your hotfix, thanks again.

Reply:
Interesting, thanks for the report/complaint.

The CI is a rather standard Jenkins with a GitHub auth plugin, so if it asks for too much to let one in - perhaps there's a way for smarter setup to require less.

As for the time - it runs different build scenarios on roughly a dozen platforms with their different built-in nittiness preferences even for same-named same-version compilers. Stopping on a first error makes little sense, as the builders made 15-20 tech-years apart choke on different things. Sometimes opposite ones, like wanting `switch/case` to cover all known values of an enum, to always have a `default` label, and disliking the label when used with an enum where all values were handled. Alas, that led to a significant amount of pragmas, albeit needed in a handful of code patterns. But, NUT still builds and works everywhere it did earlier...

Reply:
Also, in Github links to broken tests, it should have URLs to that specific failed case's logs on CI. Could you please check - do those artifacts now also require a login with "too many permissions required", or work anonymously too?

Reply:
I am not able to find back to the failed jobs any longer. But it looked like all links to the external CI required a login. There was an "aggregated" summary on GitHub saying there was a "conditional-uninitialized" warning but not specifying where.